gstreamer/src/toc.rs

```diff
@@ -256,6 +256,122 @@ impl fmt::Debug for TocEntryRef {
 unsafe impl Sync for TocEntryRef {}
 unsafe impl Send for TocEntryRef {}
 
+#[cfg(feature = "ser_de")]
+mod serde {
+    use serde::de::{Deserialize, Deserializer};
+    use serde::ser::{Serialize, Serializer, SerializeStruct};
+
+    use tags::*;
+    use super::*;
+
+    impl Serialize for TocRef {
+        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            let mut toc = serializer.serialize_struct("Toc", 3)?;
+            toc.serialize_field("scope", &self.get_scope())?;
+            toc.serialize_field("tags", &self.get_tags())?;
+            toc.serialize_field("entries", &self.get_entries())?;
+            toc.end()
+        }
+    }
+
+    impl Serialize for Toc {
+        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            self.as_ref().serialize(serializer)
+        }
+    }
+
+    impl Serialize for TocEntryRef {
+        fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+            let mut toc_entry = serializer.serialize_struct("TocEntry", 6)?;
+            toc_entry.serialize_field("entry_type", &self.get_entry_type())?;
+            toc_entry.serialize_field("uid", &self.get_uid())?;
+            toc_entry.serialize_field("start_stop", &self.get_start_stop_times())?;
+            toc_entry.serialize_field("tags", &self.get_tags())?;
+            toc_entry.serialize_field("loop_", &self.get_loop())?;
```

Comment:
Why with underscore?

Reply:
Ah I thought for the actual serialization "output" (like the field names in the JSON) we could use any string we want and let serde rename that. That's not the case, or not simple in this case? With `serde-derive` there's an attribute to do that at least

Reply:
Yes, I think I saw this before. Thanks for pointing this.